cpp/tests/test_odesecir.cpp

```diff
@@ -1434,5 +1434,66 @@ TEST_F(ModelTestOdeSecir, model_initialization_old_date)
     mio::set_log_level(mio::LogLevel::warn);
 }
 
+TEST(TestOdeSecir, set_divi_data_invalid_dates)
+{
+    mio::set_log_level(mio::LogLevel::off);
+    auto model = mio::osecir::Model<double>(1);
+    model.populations.array().setConstant(1);
+    auto model_vector = std::vector<mio::osecir::Model<double>>{model};
+
+    // Test with date before DIVI dataset was available.
+    EXPECT_THAT(mio::osecir::details::set_divi_data(model_vector, "", {1001}, {2019, 12, 01}, 1.0), IsSuccess());
+    // Assure that populations is the same as before.
+    EXPECT_THAT(print_wrap(model_vector[0].populations.array().cast<double>()),
+                MatrixNear(print_wrap(model.populations.array().cast<double>()), 1e-10, 1e-10));
+
+    // Test with data after DIVI dataset was no longer updated.
+    EXPECT_THAT(mio::osecir::details::set_divi_data(model_vector, "", {1001}, {2025, 12, 01}, 1.0), IsSuccess());
+    EXPECT_THAT(print_wrap(model_vector[0].populations.array().cast<double>()),
+                MatrixNear(print_wrap(model.populations.array().cast<double>()), 1e-10, 1e-10));
+
+    mio::set_log_level(mio::LogLevel::warn);
+}
+
+TEST_F(ModelTestOdeSecir, set_confirmed_cases_data_with_ICU)
+{
```

Comment:
This test will cover a lot of code, but is extremely weak. It only checks that some change happened, so it would pass if set_confirmed_cases_data returned all NaNs...

Reply:
Same for other set_confirmed_cases_data_with_ICU tests. Either change the name and add a description that makes clear that this test merely checks that the function does not fail catastrophically, or extend the test.

I would like it to use some handpicked case data values such that all computation pahts (i.e. value dependent ifs, assuming there are any) are covered, then check against an expected result.